### PR TITLE
[Sparse] Fix Windows.h dependency issue

### PR DESCRIPTION
**Pull request by czkkkkkk**
## Description

Fix the build fail issue when building DGL sparse on windows with Pytorch 2.1.1.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by peizhou001 — APPROVED** [review]

File: dgl_sparse/include/sparse/torch_headers.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * https://stackoverflow.com/questions/4913922/possible-problems-with-nominmax-on-visual-c.
+ */
+
+#ifndef SPARSE_TORCH_HEADERS_H_
+#define SPARSE_TORCH_HEADERS_H_
+
+#ifdef _WIN32
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif  // NOMINMAX
```
Comment (frozenbugs) [inline dgl_sparse/include/sparse/torch_headers.h:17]:
```suggestion
#ifndef NOMINMAX
#define SPARSE_TORCH_HEADERS_H_NOMINMAX
#define NOMINMAX
#endif  // NOMINMAX
```

File: dgl_sparse/include/sparse/torch_headers.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#endif  // _WIN32
+
+#include <ATen/DLConvertor.h>
+#include <c10/util/Logging.h>
+#include <torch/custom_class.h>
+#include <torch/script.h>
+
+#ifdef _WIN32
+#undef NOMINMAX
+#endif  // _WIN32
```
Comment (frozenbugs) [inline dgl_sparse/include/sparse/torch_headers.h:27]:
```suggestion
#ifdef SPARSE_TORCH_HEADERS_H_NOMINMAX
#undef NOMINMAX
#endif  // SPARSE_TORCH_HEADERS_H_NOMINMAX
```

**Review by frozenbugs — APPROVED** [review]

**Comment by frozenbugs** [comment]
@dgl-bot

File: dgl_sparse/include/sparse/torch_headers.h (hunk 1 above)
Comment (czkkkkkk) [reply dgl_sparse/include/sparse/torch_headers.h:17]: I didn't get it. What is the difference?

Comment (frozenbugs) [reply dgl_sparse/include/sparse/torch_headers.h:17]: In your implementation, #ifdef _WIN32, the NOMINMAX will be undefined regardless whether it is defined outside this header file or not. While the new one will only undefine it if the NOMINMAX by this file.

Comment (czkkkkkk) [reply dgl_sparse/include/sparse/torch_headers.h:17]: I see. Totally make sense.

**Pull request merged**